raiden/smart_contracts/ChannelManagerContract.sol

```diff
@@ -114,7 +120,11 @@ contract ChannelManagerContract {
     }
 
     function newChannel(address partner, uint settle_timeout) returns (address channel) {
-        channel = data.newChannel(msg.sender, partner, settle_timeout);
+        address old_channel;
+        bool channelDeleted;
+        (channel, channelDeleted, old_channel)  = data.newChannel(msg.sender, partner, settle_timeout);
+        // Only if channelDeleted is true raise an ChannelDeleted event.
+        if (channelDeleted) ChannelDeleted(msg.sender, partner, old_channel);
```

Comment:
There is no need for this here. It's unnecessarily complicated. Move the `ChannelDeleted` event into the `deleteChannel()` function instead. Why have it here?